daemon/daemon_unix.go

```diff
@@ -77,7 +77,7 @@ func getMemoryResources(config containertypes.Resources) *specs.Memory {
 		memory.Reservation = &reservation
 	}
 
-	if config.MemorySwap != 0 {
+	if config.MemorySwap > 0 {
```

Comment:
This routine seems to only be used when creating the container spec initially (not when updating), in which case it's correct. The default value for that limit being `unlimited`.